plugins/module_utils/dnac.py

```diff
@@ -992,6 +993,182 @@ def check_tasks_response_status(self, response, api_name):
+            self.result["data"] = data
+
+        return self
+
+    def fail_and_exit(self, msg):
+        """Helper method to update the result as failed and exit."""
+        self.update_result("failed", False, msg, "ERROR")
+        self.check_return_status()
+
+    def log_traceback(self):
+        """
+        Logs the full traceback of the current exception.
+        """
+        # Capture the full traceback
+        full_traceback = traceback.format_exc()
+        
+        # Log the traceback
+        self.log("Traceback: {0}".format(full_traceback), "DEBUG")
+
+    def exit_while_loop(self, start_time, task_id, task_name, response):
+        """
+        Check if the elapsed time exceeds the specified timeout period and exit the while loop if it does.
+        Parameters:
+            - start_time (float): The time when the while loop started.
+            - task_id (str): ID of the task being monitored.
+            - task_name (str): Name of the task being monitored.
+            - response (dict): Response received from the task status check.
+        Returns:
+            bool: True if the elapsed time exceeds the timeout period, False otherwise.
+        """
+        # If the elapsed time exceeds the timeo.,ut period
+        if time.time() - start_time > self.params.get("dnac_api_task_timeout"):
+            if response.get("data"):
+                # If there is data in the response, include it in the error message
+                self.msg = "Task {0} with task id {1} has not completed within the timeout period. Task Status: {2} ".format(
+                    task_name, task_id, response.get("data"))
+            else:
+                # If there is no data in the response, generate a generic error message
+                self.msg = "Task {0} with task id {1} has not completed within the timeout period.".format(
+                    task_name, task_id)
+
+            # Update the result with failure status and log the error message
+            self.update_result("failed", False, self.msg, "ERROR")
+            return True
+
+        return False
+
+    def get_taskid_post_api_call(self, api_family, api_function, api_parameters):
+        """
+        Executes the specified API call with given parameters and logs responses.
+
+        Parameters:
+            api_family (str): The API family (e.g., "sda").
+            api_function (str): The API function (e.g., "add_port_assignments").
+            api_parameters (dict): The parameters for the API call.
+        """
+        try:
+            self.log("Entered {0} method".format(api_function), "DEBUG")
+
+            # Execute the API call
+            response = self.dnac._exec(
+                family=api_family,
+                function=api_function,
+                op_modifies=True,
+                params=api_parameters,
+            )
+            self.log("Response received post {0} - '{1}' API call: {2}".format(api_family, api_function, str(response)), "DEBUG")
```

Comment:
To be more clear
`        self.log("API call response for '{0}' in '{1}' family: {2}".format(api_function, api_family, str(response)), "DEBUG")`
OR

`            self.log("Response received post API call '{1}' in family: {2}".format(api_family, api_function, str(response)), "DEBUG")
`